raiden/transfer/mediated_transfer/initiator_manager.py

```diff
@@ -45,7 +45,7 @@ def clear_if_finalized(
     state = cast(InitiatorPaymentState, iteration.new_state)
 
     if state is None:
-        return iteration
+        return iteration  # type: ignore
```

Comment:
Not sure what to do about this one. According to `TransitionResult`, `new_state` should never be `None`, but we use it as that sometimes.

Reply:
hummm, I forgot about this case. So the problem is:

- ChainState can never be None
- A transfer task is set to None once it is finished.

Probably the right fix is to have the transition result defined with a typevar and in the tasks use `Optional`